types/i18n.d.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Util = require('../lighthouse-core/report/html/renderer/util.js');
+
+declare global {
+  module LH {
+    export type IcuMessage = {
+      // NOTE: `i18nId` rather than just `id` to make tsc typing easier (vs type branding which won't survive JSON roundtrip).
+      /** The id locating this message in the locale message json files. */
+      i18nId: string;
+      /** The dynamic values, if any, to insert into the localized string. */
+      values?: Record<string, string | number>;
+      /**
+       * A formatted version of the string, usually as found in a file's `UIStrings`
+       * entry, and used as backup if the `i18nId` doesn't refer to an existing
+       * message in the requested locale. Used when serialized and a new version of
+       * Lighthouse no longer contains the needed message (or in development and
+       * the locale files don't yet contain it).
+       */
+      formattedDefault: string,
+    };
+
+    /**
+     * A helper to represent the type of an object that hasn't be localized yet.
+     * Recursively finds all `string` properties in `T` and extends them to also
+     * accept an `LH.IcuMessage`.
+     *
+     * Heavy handed and requires more type checks, so prefer explicitly setting
+     * properties to include `LH.IcuMessage` over this helper if possible.
+     */
+    type Icu<T> = T extends IcuMessage ? T :
```

Comment:
how do you feel about calling this `UnformattedIcu` or `RawIcu` or something?

it was a bit weird seeing this in usage until I came here to see it sitting in direct contrast to `FormattedIcu`

Reply:
> how do you feel about calling this `UnformattedIcu` or `RawIcu` or something?
> 
> it was a bit weird seeing this in usage until I came here to see it sitting in direct contrast to `FormattedIcu`

`RawIcu` sounds good to me since `UnformattedIcu` is a little visually confusable with `FormattedIcu`, but happy to bikeshed more now that it's been 5 months since this comment :)